File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int ans = INT_MAX;
        int l = 0;
        int h = nums.size()-1;
        while(l<=h)
        {
            int mid = l + (h-l)/2;
            if(nums[l]<=nums[h]) return min(ans,nums[l]);
            else if(nums[l]<=nums[mid]){
                  ans = min(ans,nums[l]);
                  l = mid+1;
            }
            else{
                ans = min(ans,nums[mid]);
                  h = mid-1;
            }
        }
        return ans;

    }
};

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int i = 0;
        int j = nums.length-1;
        int min = Integer.MAX_VALUE;
        while(i<=j)
        {
            int mid = i + (j-i)/2;
            if(nums[i]<nums[j])
            {
                min = Math.min(min,nums[i]);
                break;
            }
            if(nums[mid]<nums[j])
            {
                min = Math.min(min,nums[mid]);
                j= mid-1;
            }
            else{
                min = Math.min(min,nums[i]);
                i = mid+1;
            }
        }
        return min;
    }
}